src/cephadm/cephadm

```diff
@@ -2234,7 +2234,11 @@ def get_container_mounts(ctx, fsid, daemon_type, daemon_id,
     if daemon_type == 'osd':
         mounts['/sys'] = '/sys'  # for numa.cc, pick_address, cgroups, ...
         # selinux-policy in the container may not match the host.
-        mounts['/usr/share/empty'] = '/sys/fs/selinux:ro'
+        if HostFacts(ctx).selinux_enabled:
+            selinux_folder = '/var/lib/ceph/%s/selinux' % fsid
+            if not os.path.exists(selinux_folder):
+                os.makedirs(selinux_folder, mode=0o755)
+                mounts[selinux_folder] = '/sys/fs/selinux:ro'
```

Comment:
A small concern, here. Are we mounting the folder only if it did not exist before? Shouldn't we always mount it?

Reply:
Oops.. yeah, that looks like a misindentation

Reply:
https://github.com/ceph/ceph/pull/39490

Reply:
OMG. This change is resulting really difficult!!